Source/CarthageKit/Xcode.swift

```diff
@@ -913,6 +917,47 @@ public func buildDependencyProject(dependency: ProjectIdentifier, rootDirectoryU
 		}
 }
 
+public func signingIdentities() -> SignalProducer<String, CarthageError> {
+	let securityTask = TaskDescription(launchPath: "/usr/bin/security", arguments: ["find-identity", "-v", "-p", "codesigning"])
+	
+	return launchTask(securityTask)
+		|> ignoreTaskData
+		|> mapError { CarthageError.TaskError($0) }
+		|> map { (data: NSData) -> String in
+			return NSString(data: data, encoding: NSStringEncoding(NSUTF8StringEncoding))! as String
+		}
+		|> flatMap(.Merge) { (string: String) -> SignalProducer<String, CarthageError> in
+			return string.linesProducer |> promoteErrors(CarthageError.self)
+		}
+		|> map { (identityLine: String) -> String? in
+			var error: NSError? = nil
+			let regex = NSRegularExpression(pattern: "\".+\"", options: nil, error: &error)
+			if let match = regex?.firstMatchInString(identityLine, options: nil, range: NSRange(location: 0, length: count(identityLine))) as NSTextCheckingResult? {
+				let quotedIdentityName = (identityLine as NSString).substringWithRange(match.range)
+				let identityName = quotedIdentityName.stringByTrimmingCharactersInSet(NSCharacterSet(charactersInString: "\""))
+				return identityName
+			} else {
+				return nil
+			}
+		}
+		|> ignoreNil
+}
+
+/// Returns true if the current user has any iOS signing identities configured
+public func iOSSigningIdentitiesConfigured(identities: SignalProducer<String, CarthageError> = signingIdentities()) -> Bool {
+	let iOSIdentities = identities
+		|> filter { (identity) in
+			let id = identity as NSString
+			return id.containsString("iOS") || id.containsString("iPhone")
```

Comment:
I don't think this is the correct approach. Instead of checking that the name of the identity contains `iOS` or `iPhone`, we should cross-reference the installed signing identities with the build settings of the scheme (Carthage already has code to get the build settings).

I'm not positive which setting we need to look at, but #281 suggests that we should use `EXPANDED_CODE_SIGN_IDENTITY`.